### PR TITLE
Issue69: Use scenetree on descendant nodes

### PR DESCRIPTION
**Pull request by benjiwolff**
Allows accessing the scene tree from descendant nodes #69.

### CODE REVIEW

**Comment by benjiwolff** [comment]
I am not sure how to run the tests. There seem to be broken tests no main. However, I was able to test the generator in another project.

File: SourceGenerators/SceneTreeExtensions/SceneTreeTemplate.sbncs [hunk 1]
```diff
@@ -77,7 +77,30 @@ end
...
+{{NSIndent}}                if (sceneRoot.GetNodeOrNull(candidatePath) == this)
+{{NSIndent}}                {
+{{NSIndent}}                    return sceneRoot;
+{{NSIndent}}                }
+{{NSIndent}}            }
+{{NSIndent}}
+{{NSIndent}}            sceneRoot = sceneRoot.GetParent();
+{{NSIndent}}            if (sceneRoot is null)
+{{NSIndent}}            {
+{{NSIndent}}                throw new System.InvalidOperationException("Failed to find scene root.");
```
[the remark this answers is not in the document]
Comment (benjiwolff) [reply SourceGenerators/SceneTreeExtensions/SceneTreeTemplate.sbncs]: Done.

**Comment by Cat-Lips** [comment]
Awesome, thanks for this.  Can be tricky to find the time for new features :(

Let me know when you're happy with it, I'll merge to a branch (rather than main), run/add new tests, etc :)

**Comment by benjiwolff** [comment]
The only thing I would change, is generating the scene tree class separately from the partial script class. I'll create a PR for the feature branch.

**Comment by benjiwolff** [comment]
Here's the follow-up PR #83. I am happy with it now. 😁

**Comment by Cat-Lips** [comment]
Ok, great.  New pre-release nuget contains SceneTree on sub-node changes with type-safe root access (via Get() and implicit conversion).

**Pull request merged**